src/shared_modules/utils/rocksDBOptions.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * April 9, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _ROCKS_DB_OPTIONS_HPP
+#define _ROCKS_DB_OPTIONS_HPP
+
+#include <memory>
+#include <rocksdb/db.h>
+#include <rocksdb/table.h>
+
+namespace Utils
+{
+
+    class RocksDBOptions final
+    {
+
+        /**
+         * @brief Builds the table options for the RocksDB instance.
+         * @return rocksdb::BlockBasedTableOptions Table options.
+         */
+        static rocksdb::BlockBasedTableOptions buildTableOptions(const std::shared_ptr<rocksdb::Cache>& readCache)
+        {
+            if (readCache == nullptr)
+            {
+                throw std::runtime_error("Read cache is not initialized");
+            }
+
+            rocksdb::BlockBasedTableOptions tableOptions;
+            tableOptions.block_cache = readCache;
+            return tableOptions;
+        }
+
+    public:
+        /**
+         * @brief Builds the column family options for the RocksDB instance.
+         * @return rocksdb::ColumnFamilyOptions Column family options.
+         */
+        static rocksdb::ColumnFamilyOptions buildColumnFamilyOptions(const std::shared_ptr<rocksdb::Cache>& readCache)
+        {
+            rocksdb::ColumnFamilyOptions columnFamilyOptions;
+            // Amount of data to build up in memory (backed by an unsorted log
+            // on disk) before converting to a sorted on-disk file.
+            columnFamilyOptions.write_buffer_size = 64 * 1024 * 1024;
```

Comment:
We can avoid the calculation in runtime

```cpp
constexpr uint32_t BUFFER_SIZE = 64 * 1024 * 1024
```


```suggestion
            columnFamilyOptions.write_buffer_size = BUFFER_SIZE;
```